lib/literal/failure.rb

```diff
@@ -1,6 +1,13 @@
 # frozen_string_literal: true
 
 class Literal::Failure < Literal::Result
+	DECONSTRUCTION_TEMPLATE = {
+		message: -> (error) { error.message },
+		exception: -> (error) { error.exception },
+		full_message: -> (error) { error.full_message },
+		detailed_message: -> (error) { error.detailed_message }
+	}
```

Comment:
@stevegeek what are your thoughts on this? Exception itself doesn't have pattern matching and I wanted to allow pattern matching on some of the main attributes without always calling them.

Reply:
Yeah nice that you can match on elements of the internal error. I would have expected that Exception implements the deconstruct methods itself, I guess for some reason they do not...

One comment, would you want to support other instance methods of the exception as matchable properties, `:detailed_message, :backtrace, :backtrace_locations, :cause, :full_message, :message, :exception` I think is the full set

For example https://github.com/joeldrapper/literal/pull/49